src/components/modal/modal.js

```diff
@@ -137,7 +137,8 @@ class Modal extends mixin(createComponent, initComponentByLauncher, eventedShowH
 
     this._handleKeydownListener = this.manage(
       on(this.element.ownerDocument.body, 'keydown', evt => {
-        if (evt.which === 27) {
+        // Avoid running `evt.stopPropagation()` only when modal is shown
+        if (evt.which === 27 && this.shouldStateBeChanged('hidden')) {
```

Comment:
do we want to use our keyboard event helpers here?

Reply:
@emyarod Which helper do you have in mind?

Reply:
none, I initially thought we had the same helpers as we do in the React repo 😅